rust/nix-expr/src/eval_state.rs

```diff
@@ -191,12 +237,24 @@ impl Drop for EvalState {
 #[cfg(test)]
 mod tests {
     use ctor::ctor;
+    use nix_util::settings;
 
     use super::*;
 
     #[ctor]
     fn setup() {
-        init().unwrap();
+        (|| -> Result<()> {
+            init()?;
+            // If it reinvokes the test suite,
+            // settings::set("build-hook", "")?;
+
+            // When testing in the sandbox, the default build dir would be a parent of the storeDir,
+            // which causes an error. So we set a custom build dir here.
+            settings::set("sandbox-build-dir", "/custom-build-dir-for-test")?;
+            Ok(())
+        })()
+        .unwrap();
+        std::env::set_var("_NIX_TEST_NO_SANDBOX", "1");
```

Comment:
This too. Is this affecting the building **of this repo**? And if so should it be in the .nix files?

Reply:
See https://github.com/nixops4/nixops4/pull/17#discussion_r1609794003